lib/ejs.js

```diff
@@ -399,17 +418,27 @@ exports.render = function (template, d, o) {
  */
 
 exports.renderFile = function () {
-  var filename = arguments[0];
-  var cb = arguments[arguments.length - 1];
+  var args = Array.prototype.slice.call(arguments);
+  var filename = args.shift();
+  var cb;
   var opts = {filename: filename};
   var data;
 
-  if (arguments.length > 2) {
-    data = arguments[1];
-
-    // No options object -- if there are optiony names
-    // in the data, copy them to options
-    if (arguments.length === 3) {
+  // Do we have a callback?
+  if (typeof arguments[arguments.length - 1] == 'function') {
```

Comment:
`===`?

Reply:
See comments above. :)

Reply:
@niftylettuce === is totally correct == is weak and don't checks for type so === checks including type in this case number if you want to have a Javascript Training or need any other Consulting simpy contact me i am free for individual trainings

Reply:
@mde sorry saw it to late you should do === not == for same reason as above!

Reply:
Please see above. Thanks for the comments, but I will not be applying the strict-equality operator everywhere.